prqlc/prql-compiler/tests/integration/queries.rs

```diff
@@ -58,23 +58,68 @@ mod results {
     use once_cell::sync::Lazy;
     use prql_compiler::sql::SupportLevel;
 
-    use crate::dbs::{ConnectionCfg, DbConnection};
+    use crate::dbs::{ConnectionCfg, DbConnection, DbProtocol};
 
     static CONNECTIONS: Lazy<Mutex<Vec<DbConnection>>> = Lazy::new(init_connections);
 
     fn init_connections() -> Mutex<Vec<DbConnection>> {
-        let con_dir = "./tests/integration/dbs/connections";
+        let configs = [
+            ConnectionCfg {
+                dialect: Dialect::SQLite,
+                data_file_root: "tests/integration/data/chinook".to_string(),
+
+                protocol: DbProtocol::SQLite,
+            },
+            ConnectionCfg {
+                dialect: Dialect::DuckDb,
+                data_file_root: "tests/integration/data/chinook".to_string(),
+
+                protocol: DbProtocol::DuckDb,
+            },
+            ConnectionCfg {
+                dialect: Dialect::Postgres,
+                data_file_root: "/tmp/chinook".to_string(),
+
+                protocol: DbProtocol::Postgres {
+                    url: "host=localhost user=root password=root dbname=dummy".to_string(),
+                },
+            },
+            ConnectionCfg {
+                dialect: Dialect::MySql,
+                data_file_root: "/tmp/chinook".to_string(),
+
+                protocol: DbProtocol::MySql {
+                    url: "mysql://root:root@localhost:3306/dummy".to_string(),
+                },
+            },
+            ConnectionCfg {
+                dialect: Dialect::ClickHouse,
+                data_file_root: "chinook".to_string(),
+
+                protocol: DbProtocol::MySql {
+                    url: "mysql://default:@localhost:9004/dummy".to_string(),
+                },
+            },
+            ConnectionCfg {
+                dialect: Dialect::GlareDb,
+                data_file_root: "/tmp/chinook".to_string(),
+
+                protocol: DbProtocol::Postgres {
+                    url: "host=localhost user=glaredb dbname=glaredb port=6543".to_string(),
+                },
+            },
+            ConnectionCfg {
+                dialect: Dialect::MsSql,
+                data_file_root: "/tmp/chinook".to_string(),
+
+                protocol: DbProtocol::MsSql,
+            },
+        ];
```

Comment:
Initial idea was to find a test generator macro that I would point at two directories and it would then generate a test for each combination of the two files.